src/libs/actions/IOU.ts

```diff
@@ -1782,6 +1791,17 @@ function getDeleteTrackExpenseInformation(
         },
     ];
 
+    if (shouldDeleteTransactionThread && transactionThread) {
+        successData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${transactionThreadID}`,
+            value: Object.keys(transactionThread).reduce<Record<string, null>>((acc, key) => {
+                acc[key] = null;
+                return acc;
+            }, {}),
+        });
+    }
+
```

Comment:
Why is this needed?

Reply:
@luacmartins Ensure that any remaining data is removed upon successful completion, even if the server sends a report removal response. This is done to prevent the removal update from lingering in the `applyHTTPSOnyxUpdates` function. https://github.com/Expensify/App/pull/50846#discussion_r1802366119

Reply:
Same here, let's add a comment about it. We should add a comment for all instances to make sure people don't revert this

Reply:
done